belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/OpenApiRequestViolationProblemAdvice.java

```diff
@@ -0,0 +1,198 @@
+package io.github.belgif.rest.problem.spring;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import com.atlassian.oai.validator.report.ValidationReport;
+import com.atlassian.oai.validator.springmvc.InvalidRequestException;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.github.belgif.rest.problem.BadRequestProblem;
+import io.github.belgif.rest.problem.ResourceNotFoundProblem;
+import io.github.belgif.rest.problem.api.InEnum;
+import io.github.belgif.rest.problem.api.InputValidationIssue;
+import io.github.belgif.rest.problem.api.Problem;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+@RestControllerAdvice
+@ConditionalOnWebApplication
+@Order(1)
+@ConditionalOnClass(InvalidRequestException.class)
+public class OpenApiRequestViolationProblemAdvice {
```

Comment:
Can you add javadoc on the class to explain its purpose? 
Also add comment why `@Order(1)`

Reply:
Strange, I removed it locally to see if it was really needed, and the build succeeded.
But on the github build it fails. So I guess if multiple exception handlers have the same priority, the order is undetermined.